ofrak_core/ofrak/core/apk.py

```diff
@@ -127,16 +131,27 @@ async def pack(
         temp_flush_dir = await apk.flush_to_disk()
         apk_suffix = ".apk"
         with tempfile.NamedTemporaryFile(suffix=apk_suffix) as temp_apk:
-            command = ["apktool", "build", "--force-all", temp_flush_dir, "--output", temp_apk.name]
-            subprocess.run(command, check=True, capture_output=True)
+            # command = ["apktool", "build", "--force-all", temp_flush_dir, "--output", temp_apk.name]
+            apk_proc = await asyncio.create_subprocess_exec(
+                "apktool",
+                "build",
+                "--force-all",
+                temp_flush_dir,
+                "--output",
+                temp_apk.name,
+                stderr=asyncio.subprocess.PIPE,
+            )
+            stdout, stderr = await apk_proc.communicate()
+            if apk_proc.returncode and apk_proc.returncode < 0:
+                raise Exception(stderr.decode())
```

Comment:
In general, it is considered bad practice to `raise Exception` instead of a more specific subclass of `Exception` such as `RuntimeError`. 

Additionally, OFRAK components are built to handle `CalledProcessError` exceptions raised by `subprocess.run`, and should be extended to handle whatever specific exception type is raised here.

https://github.com/redballoonsecurity/ofrak/blob/c0b0cfd95f17543397224bf78bf6371c8ee6e166/ofrak_core/ofrak/component/abstract.py#L93-L103

This comment applies to all of the places throughout this PR where changes to `asyncio.create_subprocess_exec` were made.

Reply:
I think it's simplest to raise a `CalledProcessError` here, for consistency with the previous behavior